specs/APIReview_AllowDrop.md

```diff
@@ -0,0 +1,103 @@
+# Background
+Currently the drag&drop functionality is default enabled in webview2 and there is no way to disable it. Some developers may want to disbale this functionality in their applications 
+based upon their scenarios. According to such feature requirements, we add the new API to provide developers with the capability to configure the drag&drop functionality.
+
+# Description
+We add a new `AllowDrop` property in `CoreWebView2Controller`. 
```

Comment:
Why the Controller rather than WebView2? (There are properties on WV2 that configure behavior, like CanGoBack.)

Reply:
This property will be wrapped in .NET control since WPF and WinForms control of WebView2 have AllowDrop property as well. It's usually set up when creating WebView, the similar one of this property is the IsVisible property that is also in Controller and is wrapped in .NET control.

Reply:
I guess I don't understand the philosophy about what goes to which. A property should go on the Controller if it needs to be set before accessing the WV? Can it still be changed after the WV has been fetched the first time?

Reply:
The CoreWebView2Controller contains the hosting specific parts of the webview and the CoreWebView2 contains the web specific parts of the webview. The idea is that a UI framework specific WebView2 control has a CoreWebView2Controller and then exposes as a property the corresponding CoreWebView2. This way the WebView2 can expose the hosting concepts supported by the CoreWebView2Controller in a manner consistent with the WebView2's UI framework and doesn't have to keep its UI framework APIs synchronized with changes made directly on the CoreWebView2Controller.

```
WebView2      -------------
    |                      |
 (contains)             (exposes)
    |                      |
    v                      v
CoreWebView2Controller -> CoreWebView2
```

In this case allowing drag and drop is a hosting concept that WPF and other UI frameworks support and so we put the AllowDrop property on the CoreWebView2Controller so that the WebView2 controls can hook them up to their own APIs for allowing drag and drop.

Reply:
Thanks